### PR TITLE
Link for real

### PR DESCRIPTION
**Pull request by sinkingsugar**

### CODE REVIEW

**Comment by sinkingsugar** [comment]
@aang114 @alebaffa  any update here?

**Comment by alebaffa** [comment]
@aang114 said he was working on this. I created an issue for the next sprint for this too.

**Comment by aang114** [comment]
> @aang114 @alebaffa any update here?

Yup,  the message is now encoded using the EIP-712 encoding function (you can read the comments in `accounts.link()` for more info on how I encoded it). I also **manually verified** that the encoded message using `accounts.link()` matches the message I generated using this JS library:  with https://www.npmjs.com/package/eip-712/v/0.4.0?activeTab=readme

Also btw @sinkingsugar - there was no need for removing the first element of the the recovered ECDSA public key in `accounts.link()` - **since the recovered public key was 64 bytes** (https://ethereum.org/en/developers/docs/accounts/#account-creation). 

TLDR: The PR is good to go/merge I think.

File: pallets/accounts/src/lib.rs [hunk 1]
```diff
@@ -385,16 +389,71 @@ pub mod pallet {
...
+			// We emulate JS JSON.stringify behavior here basically...
+			// let message = format!(r#"{{"domain":{{"name":"Fragnova Network","version":"1","chainId":{},"verifyingContract":"0xF5A0Af5a0AF5a0AF5a0af5A0Af5A0AF5a0AF5A0A"}},"message":{{"fragnovaGenesis":"{}","op":"link","sender":"{}"}},"primaryType":"Msg","types":{{"EIP712Domain":[{{"name":"name","type":"string"}},{{"name":"version","type":"string"}},{{"name":"chainId","type":"uint256"}},{{"name":"verifyingContract","type":"address"}}],"Msg":[{{"name":"fragnovaGenesis","type":"string"}},{{"name":"op","type":"string"}},{{"name":"sender","type":"string"}}]}}}}"#, T::EthChainId::get(), genesis_hash, sender_string);
+
+			// TODO, THE ABOVE IS WRONG.
+			// We need to follow https://eips.ethereum.org/EIPS/eip-712 spec.. which is convoluted...
+
+			// We encode the message using the following encoding function:
+			// encode(domainSeparator : 𝔹²⁵⁶, message : 𝕊) = "\x19\x01" ‖ domainSeparator ‖ hashStruct(message).
+			// See: https://eips.ethereum.org/EIPS/eip-712#specification-1
+			let message: Vec<u8> = [
```
Comment (sinkingsugar) [inline pallets/accounts/src/lib.rs]: Love it but can you write a shared common method so it can be used around the projects and tests? Thanks

Comment (sinkingsugar) [reply pallets/accounts/src/lib.rs]: @aang114

Comment (aang114) [reply pallets/accounts/src/lib.rs]: > Love it but can you write a shared common method so it can be used around the projects and tests? Thanks

A common method that returns a link message, right?

Comment (sinkingsugar) [reply pallets/accounts/src/lib.rs]: yep

**Comment by alebaffa** [comment]
@aang114 I think this one is ready, right (after resolved conflicts)?

**Comment by aang114** [comment]
@sinkingsugar , can I merge this?
